src/libConsensus/ConsensusUser.h

```diff
@@ -38,6 +39,9 @@ class ConsensusUser : public Executable, public Broadcastable
     bool m_leaderOrBackup; // false = leader, true = backup
     std::shared_ptr<ConsensusCommon> m_consensus;
 
+    std::shared_timed_mutex m_mutexProcessConsensusMessage;
```

Comment:
Why use shared_timed_mutex to only limit writer without reader?

Reply:
To allow other threads to use m_consensusObject for read purposes (in this case, using CanProcessMessage() ).  But it is still vital to ensure m_consensusObject data is not overwritten (eg reset).

Reply:
noted.